src/Stack/Setup.hs

```diff
@@ -1085,14 +1085,19 @@ installGHCPosix version downloadInfo _ archiveFile archiveType tempDir destDir =
 
     let runStep step wd env cmd args = do
             menv' <- modifyEnvOverride menv (Map.union env)
-            result <- withWorkingDir wd
-                    $ withEnvOverride menv'
-                    $ withProc cmd args
-                    $ try
-                    . readProcessStdout_
-                    -- Calling the ./configure script requires that stdin is
-                    -- open
-                    . setStdin (useHandleOpen stdin)
+            result <- do
+                let logLines = CL.mapM_ (logDebug . T.decodeUtf8With T.lenientDecode)
+                withWorkingDir wd
+                  $ withEnvOverride menv'
+                  $ withProc cmd args
+                  $ try
+                  . (flip withLoggedProcess_ $ \p ->
```

Comment:
What's the reason for the switch to `withLoggedProcess_` here, instead of say `withProcess_`? Quoting from the docs:

> This will store all output from stdout and stderr in memory for better error messages.

It seems like that's redundant with turning on the verbose debugging here.

Reply:
I shall update the PR using withProcess_.